js_modules/dagit/packages/core/src/ticks/SensorDryRun.tsx

```diff
@@ -0,0 +1,361 @@
+import {gql, useMutation} from '@apollo/client';
+import {
+  Box,
+  Button,
+  ButtonLink,
+  Colors,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Group,
+  Icon,
+  Spinner,
+  Subheading,
+  Tag,
+  TextInput,
+} from '@dagster-io/ui';
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {showCustomAlert} from '../app/CustomAlertProvider';
+import {SharedToaster} from '../app/DomUtils';
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {PythonErrorInfo} from '../app/PythonErrorInfo';
+import {assertUnreachable} from '../app/Util';
+import {PythonErrorFragment} from '../app/types/PythonErrorFragment.types';
+import {SET_CURSOR_MUTATION} from '../sensors/EditCursorDialog';
+import {
+  SetSensorCursorMutation,
+  SetSensorCursorMutationVariables,
+} from '../sensors/types/EditCursorDialog.types';
+import {testId} from '../testing/testId';
+import {RepoAddress} from '../workspace/types';
+
+import {RunRequestTable} from './DryRunRequestTable';
+import {SensorDryRunMutation, SensorDryRunMutationVariables} from './types/SensorDryRun.types';
+
+type DryRunInstigationTick = Extract<
+  SensorDryRunMutation['sensorDryRun'],
+  {__typename: 'DryRunInstigationTick'}
+>;
+
+export const SensorDryRun: React.FC<{
```

Comment:
It'd be nice to call this SensorDryRunDialog or otherwise indicate that it's primary rendered thing is a DIalog :-)